exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -791,6 +799,43 @@ public FileGroupScan clone(FileSelection selection) throws IOException {
   }
 
   @Override
+  public GroupScan applyLimit(long maxRecords) {
```

Comment:
I gave some thoughts about this optimization as well. Then, I realized that until we have some performance measurement, it's not very clear which way we want to have. For example, I'm not clear whether 1000 small parquet files is better than 1 large parquet files. 1000 files might have big metadata overhead than 1 large file (?). But 1000 small files might be better option, in case we do want to parallelize the execution.

I'll add some comment saying further optimization could be done in terms of how subset of files are chosen.